### PR TITLE
Invoca 4.2.11

### PR DESCRIPTION
**Pull request by alecjacobs5401**
This is a PR to show the changes in getting the latest upstream version of `4-2-stable`, which was the bump to version `4.2.11`

Here is a comparison for this branch vs the upstream `4-2-stable`
https://github.com/rails/rails/compare/4-2-stable...Invoca:invoca-4.2.11?expand=1

### CODE REVIEW

**Review by jebentier — APPROVED** [review]
looks pretty harmless to me

File: Gemfile [hunk 1]
```diff
@@ -20,7 +20,12 @@ gem 'mocha', '~> 0.14', require: false
 gem 'rack-cache', '~> 1.2'
 gem 'jquery-rails', '~> 4.0'
 gem 'coffee-rails', '~> 4.1.0'
-gem 'turbolinks'
+
+if RUBY_VERSION < "2.1"
```
Comment (mikeweaver) [inline Gemfile:24]: We don't run with Ruby 2.1, so... why do we need this?

File: Gemfile.lock [hunk 2]
```diff
@@ -108,102 +108,107 @@ GEM
...
+    hitimes (1.2.6)
+    hitimes (1.2.6-x86-mingw32)
+    i18n (0.9.5)
+      concurrent-ruby (~> 1.0)
     jquery-rails (4.3.1)
       rails-dom-testing (>= 1, < 3)
       railties (>= 4.2.0)
       thor (>= 0.14, < 2.0)
-    json (2.1.0)
+    json (1.8.6)
```
Comment (mikeweaver) [inline Gemfile.lock:133]: JSON got downgraded?

File: Gemfile.lock [hunk 3]
```diff
@@ -108,102 +108,107 @@ GEM
...
-    rdoc (5.1.0)
+    rails-html-sanitizer (1.0.4)
+      loofah (~> 2.2, >= 2.2.2)
+    rake (12.2.1)
+    rdoc (4.3.0)
     redcarpet (3.1.2)
-    redis (3.3.3)
-    redis-namespace (1.5.3)
-      redis (~> 3.0, >= 3.0.4)
+    redis (4.0.1)
```
Comment (mikeweaver) [inline Gemfile.lock:184]: Major version bump for redis. Seems risky. Did you look at the change set?

File: Gemfile.lock [hunk 4]
```diff
@@ -108,102 +108,107 @@ GEM
...
       nokogiri (~> 1.6)
       rails-deprecated_sanitizer (>= 1.0.1)
-    rails-html-sanitizer (1.0.3)
-      loofah (~> 2.0)
-    rake (12.0.0)
-    rdoc (5.1.0)
+    rails-html-sanitizer (1.0.4)
+      loofah (~> 2.2, >= 2.2.2)
+    rake (12.2.1)
+    rdoc (4.3.0)
```
Comment (mikeweaver) [inline Gemfile.lock:182]: rdoc downgrade. Odd?

File: Gemfile.lock [hunk 5]
```diff
@@ -108,102 +108,107 @@ GEM
...
-      rdoc (~> 5.0)
-    sequel (4.47.0)
+    sdoc (0.4.2)
+      json (~> 1.7, >= 1.7.7)
+      rdoc (~> 4.0)
+    sequel (5.6.0)
     serverengine (1.5.11)
       sigdump (~> 0.2.2)
-    sidekiq (4.2.10)
+    sidekiq (5.1.3)
```
Comment (mikeweaver) [inline Gemfile.lock:207]: Major version bump here too?

**Review by mikeweaver — COMMENTED** [review]
I found a few oddities. They a probably nothing, but something to keep an eye out for the web gem file after making the upgrade.

File: Gemfile (hunk 1 above)
Comment (alecjacobs5401) [reply Gemfile:24]: I was just merging in the latest stable of `4.2`, which is `4.2.11`.

I didn't see a reason not to include everything that was in `4.2.11`

File: Gemfile.lock (hunk 2 above)
Comment (alecjacobs5401) [reply Gemfile.lock:133]: Yeah, there were some internal Rails gem changes.

Nothing in the web repo was affected

File: Gemfile.lock (hunk 3 above)
Comment (alecjacobs5401) [reply Gemfile.lock:184]: Yeah I thought this was interesting too, but it isn't a hard dependency for rails so it shouldn't affect us

File: Gemfile.lock (hunk 5 above)
Comment (alecjacobs5401) [reply Gemfile.lock:207]: This is actually the version of sidekiq we run in web, so I wasn't worried about this change

**Review by mikeweaver — APPROVED** [review]

**Comment by alecjacobs5401** [comment]
@mikeweaver 
All valid points on the gem changes, I had the same thoughts when I noticed they were pulled in on merge

**Comment by alecjacobs5401** [comment]
no need to keep this PR around, closing it

**Pull request closed without merging**